### PR TITLE
[#4874] Raise OSError on setting permission instead of AttributeError.

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
=====

This updated the Windows behaviour to look a bit closer to Unix when setting owner and group.

Changes
=======

Always raise OSError

As a drive by I created separeate tests for code which has different behaviour on Linux and Windows.

As a drive-by change I wrote a test for setAttribute for time uid and mode.

How to try and test the changes
===============================

reviewers: @dumol 

@dumol  Check to see if you can make sense of the code.

I would also like to use this PR to make sure that compat test are green in order to support the python-package.

I am also updating the openbsd part.

### CODE REVIEW

**Comment by adiroiban** [comment]
needs-review

File: chevah/compat/tests/normal/test_filesystem.py [hunk 1]
```diff
@@ -101,15 +101,13 @@ def test_getPath_upper_paths(self):
         self.assertEqual(u'/c', path)
 
 
-class TestLocalFilesystem(CompatTestCase, FilesystemTestMixin):
+class DefaulFilesystemTestCase(CompatTestCase, FilesystemTestMixin):
```
Comment (dumol) [inline chevah/compat/tests/normal/test_filesystem.py]: Curious why this is not `DefaultFilesystemTestCase`…

**Comment by dumol** [comment]
**changes-approved**

File: chevah/compat/tests/normal/test_filesystem.py (hunk 1 above)
Comment (adiroiban) [reply chevah/compat/tests/normal/test_filesystem.py]: typo :) Will Fix. Thanks!

**Pull request merged**